chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,296 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+            conds = eve_fullpuz.run(eve_sol)
+            eve_puzzle_announcement = [x for x in conds.as_python() if int_from_bytes(x[0]) == 62][0][1]
+            assertion = std_hash(eve_fullpuz.get_tree_hash() + eve_puzzle_announcement)
+            puzzle_assertions.add(assertion)
+
+        # We've now created all the intermediate, launcher, eve and transfer spends.
+        # Create the xch spend to fund the minting.
+        spend_value = sum([coin.amount for coin in xch_coins])
+        change: uint64 = uint64(spend_value - total_amount)
+        xch_spends = []
+        if xch_change_ph is None:
+            xch_change_ph = await self.standard_wallet.get_new_puzzlehash()
+        xch_primaries = [
+            AmountWithPuzzlehash({"puzzlehash": xch_change_ph, "amount": change, "memos": [xch_change_ph]})
+        ]
+
+        first = True
+        for xch_coin in xch_coins:
+            puzzle: Program = await self.standard_wallet.puzzle_for_puzzle_hash(xch_coin.puzzle_hash)
+            if first:
+                message_list: List[bytes32] = [c.name() for c in xch_coins]
+                message_list.append(
+                    Coin(xch_coin.name(), xch_primaries[0]["puzzlehash"], xch_primaries[0]["amount"]).name()
+                )
+                message: bytes32 = std_hash(b"".join(message_list))
+
+                if len(xch_coins) > 1:
+                    xch_announcement: Optional[Set[bytes]] = {message}
+                else:
+                    xch_announcement = None
+
+                solution: Program = self.standard_wallet.make_solution(
+                    primaries=xch_primaries,
+                    fee=fee,
+                    coin_announcements=xch_announcement,
+                    coin_announcements_to_assert={Announcement(did_coin.name(), message).name()},
+                )
+                primary_announcement_hash = Announcement(xch_coin.name(), message).name()
+                # connect this coin assertion to the DID announcement
+                did_coin_announcement = {bytes(message)}
+                first = False
+            else:
+                solution = self.standard_wallet.make_solution(
+                    primaries=[], coin_announcements_to_assert={primary_announcement_hash}
+                )
+            xch_spends.append(CoinSpend(xch_coin, puzzle, solution))
+        xch_spend = await self.standard_wallet.sign_transaction(xch_spends)
+
+        # Create the DID spend using the announcements collected when making the intermediate launcher coins
+        did_p2_solution = self.standard_wallet.make_solution(
+            primaries=primaries,
+            puzzle_announcements=set(launcher_ids),
+            coin_announcements=did_coin_announcement,
+            coin_announcements_to_assert=did_announcements,
+            puzzle_announcements_to_assert=puzzle_assertions,
+        )
+        did_inner_sol: Program = Program.to([1, did_p2_solution])
```

Comment:
This must be working, but for future reference I think this should be `(1, did_p2_solution)` which is the difference between `(q . foo)` and `(q foo)`